VACMS 19548 - orientation of map and controls for Facility Locator

eselkin opened the pull request:
DO NOT MERGE INTO MAIN - this is the base for the integration of changes for FL for staging review

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Updates FL on small desktop sizes and up to match the new Figma designs
- Updates to use va-button and other vads components
- Sitewide team

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19548

## Testing done

- Updates tests to match new design on desktop

## Screenshots

<details>
<summary>New Small Desktop Display</summary>
<img src="https://github.com/user-attachments/assets/e01a319c-2990-434c-b82f-ad07a294f1cb">
</details>

<details>
<summary>Small Desktop Display Old</summary>
<img src="https://github.com/user-attachments/assets/370c1f2c-d0a9-4bc5-a979-105442f44ca6">
</details>


<details>
<summary>Tablet new (slight changes due to component changes)</summary>
<img src="https://github.com/user-attachments/assets/8809254c-0888-4772-958e-5da3331c7d8e">
</details>


<details>
<summary>Tablet old</summary>
<img src="https://github.com/user-attachments/assets/e3a2ddfd-edc5-4775-ad52-d9041f743006">
</details>

Mobile has similar changes to tablet (using of the va-button makes it stuck in size despite being mobile - known issue)
<details>
<summary>Mobile map (new)</summary>
<img src="https://github.com/user-attachments/assets/72d10364-384f-4733-84dd-530f72c2bf06">
</details>

<details>
<summary>Mobile results (new)</summary>
<img src="https://github.com/user-attachments/assets/8c8925e7-be95-4aa7-a650-3d4f0439d6d5">
</details>


<details>
<summary>Mobile Map (old)</summary>
<img src="https://github.com/user-attachments/assets/243f53b3-c64b-4580-a15d-efac23019081">
</details>

<details>
<summary>Mobile Results (old)</summary>
<img src="https://github.com/user-attachments/assets/f95dbf4b-ab24-4482-8f79-8ec63cd901a7">
</details>





## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

Check /find-locations
Check on all viewport sizes
Check for all facility types and representative number of services

eselkin committed:
update controls

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
fix resize

eselkin committed:
fix radius

eselkin committed:
cleanup lock

eselkin committed:
fix

eselkin committed:
min radius is now 60 so >=

eselkin commented on config/webpack.dev.config.js:
```diff
@@ -52,7 +52,7 @@ function generateWebpackDevConfig(buildOptions) {
     port: buildOptions.port,
     host: buildOptions.host,
-    client: { webSocketURL },
+    client: { webSocketURL, overlay: false },
```
Allows some tests to run if using `yarn watch`

eselkin commented on src/applications/facility-locator/components/ControlResultsHolder.jsx:
```diff
@@ -0,0 +1,18 @@
+function ControlResultsHolder({ children, isSmallDesktop }) {
+  // If the screen is smaller than small desktop we just return the children
+  if (!isSmallDesktop) {
+    return children;
```
Cannot be enclosed in a div if not small desktop or smaller because grouping and orientation is completely different.

eselkin commented on src/applications/facility-locator/components/SearchControls.jsx:
```diff
@@ -68,6 +69,7 @@ const SearchControls = props => {
 
   const handleSubmit = e => {
     e.preventDefault();
+    e.stopPropagation();
```
Added because of va-button effects, but also limited with the type of submission from the button.

eselkin committed:
fix zip

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin commented on src/applications/facility-locator/containers/FacilitiesMap.jsx:
```diff
@@ -628,6 +698,7 @@ const FacilitiesMap = props => {
   useEffect(() => {
     setMap(setupMap());
     setUpResizeEventListener();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
This was in the code already, but just made the comment to comply

randimays reviewed (commented):
I think these two ACs still need to be addressed:

- Text for address entry updated to "Zip code or city, state"
- "Title" attribute is removed from the "Zip code, or city, state" input field

On mobile, the Search button should be full width. It looks like we also have too much left/right padding on the current implementation (gray background).

| [Figma](https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=963-20551&t=NFIUlv1IHCnzKSsY-0) | RI |
| --- | --- |
| <img width="200" alt="Screenshot 2024-12-26 at 2 23 58 PM" src="https://github.com/user-attachments/assets/41f21270-d7d0-4ce6-9feb-58a60b8aa8af" /> | <img width="200" alt="Screenshot 2024-12-26 at 2 25 23 PM" src="https://github.com/user-attachments/assets/1988a5da-493c-4b7e-a991-2f25bfbe5625" /> |

I noticed that when it's in the error state, the zip code field isn't full width like the other fields

<img width="482" alt="Screenshot 2024-12-26 at 2 55 32 PM" src="https://github.com/user-attachments/assets/ce9b8d53-b46d-4579-b865-028556b3b6c0" />

There are some other screen widths where the fields are not the same length. This one is at 600px.

<img width="603" alt="Screenshot 2024-12-26 at 2 57 59 PM" src="https://github.com/user-attachments/assets/495a94f5-e889-4865-ba50-1b950d5b5bb6" />

eselkin commented on src/applications/facility-locator/containers/FacilitiesMap.jsx:
```diff
@@ -435,47 +447,81 @@ const FacilitiesMap = props => {
+                  pagination={pagination}
+                />
+              )}
+              {searchError && <p />}
```
I was trying to track down why and seems to be a spacing thing?

eselkin replied:
You'd think it could have been handled in styles if it was just a spacing thing, though

randimays replied:
Yeah, it's not great for accessibility to have empty markup :\

eselkin replied:
Evidently it throws off that column thing with the different viewports if you take out the empty paragraph. It's like it's there for clearing.

randimays replied:
Very odd.

eselkin commented on src/applications/facility-locator/containers/FacilitiesMap.jsx:
```diff
@@ -435,47 +447,81 @@ const FacilitiesMap = props => {
+              clearSearchText={props.clearSearchText}
             />
+            <EmregencyCareAlert
+              shouldShow={isEmergencyCareType || isCppEmergencyCareTypes}
```
Yes, probably. Will change

eselkin commented on src/applications/facility-locator/sass/facility-locator.scss:
```diff
@@ -86,7 +95,7 @@ $facility-locator-shadow: rgba(0, 0, 0, 0.5);
 
   @media screen and (max-width: $small-screen) {
-    .facility-result + .facility-result {
+    .facility-result+.facility-result {
```
Yes, it's auto-linting and forcing it. I can add the spaces back

eselkin commented on src/applications/facility-locator/components/EmergencyCareAlert.jsx:
```diff
@@ -0,0 +1,28 @@
+import PropTypes from 'prop-types';
+import { VaAlert } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+
+export default function EmregencyCareAlert({ shouldShow = false }) {
```
Oops, thanks

eselkin replied:
fixed

eselkin commented:
> I think these two ACs still need to be addressed:
> 
> * Text for address entry updated to "Zip code or city, state"
> * "Title" attribute is removed from the "Zip code, or city, state" input field

Hrmm, seems my changes didn't get checked in. Will update.

> On mobile, the Search button should be full width. It looks like we also have too much left/right padding on the current implementation (gray background).

On mobile, there's an issue with the va-button not extending to full-width. Jordan was ok with it not extending until they fix it. Also Jordan was ok with it not extending on other displays until after we bring it up on Jan 16th (if time permits) in the DST/design meeting

> I noticed that when it's in the error state, the zip code field isn't full width like the other fields
This needs to get fixed. Will fix

> There are some other screen widths where the fields are not the same length. This one is at 600px.

This is a known issue and Jordan is ok with this, unfortunately, seems to be related to the va-select. Also bringing up this issue in DST - Jan 16.

eselkin commented on src/applications/facility-locator/sass/facility-locator.scss:
```diff
@@ -516,9 +560,15 @@ $facility-locator-shadow: rgba(0, 0, 0, 0.5);
 
 .mapboxgl-ctrl-top-center {
-  position: absolute;
+  // position: absolute;
```
Removed.

eselkin committed:
style updates

eselkin commented on src/applications/facility-locator/sass/facility-locator.scss:
```diff
@@ -29,20 +29,29 @@ $facility-locator-shadow: rgba(0, 0, 0, 0.5);
   .desktop-container {
     margin-bottom: 3em;
     border-bottom: solid thin var(--vads-color-base-lightest);
+
```
Removed.

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
fix widths on va-button - for future of configurable width

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
does this test work both locally and on CI

eselkin committed:
manage demensions

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
more configurable sizes for CI

eselkin committed:
.25

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
more mobile test issues that are not really mobile

eselkin committed:
merge

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

randimays reviewed (commented):
I have a couple more observations:

1. The vertical spacing between the labels for the inputs and the inputs themselves is a bit inconsistent. Can we make these the same size?

<img width="402" alt="Screenshot 2025-01-07 at 11 35 37 AM" src="https://github.com/user-attachments/assets/82c5c5e3-5a46-40c3-8eba-7fa3c5d0b19f" />

2. We do still need to address the padding on the gray background as it doesn't quite match the Figma. It should be narrower on the left and right and a bit wider on the top and bottom.

| [Figma](https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=963-20551&t=NFIUlv1IHCnzKSsY-0) | RI |
| --- | --- |
| <img width="200" alt="Screenshot 2024-12-26 at 2 23 58 PM" src="https://github.com/user-attachments/assets/41f21270-d7d0-4ce6-9feb-58a60b8aa8af" /> | <img width="200" alt="Screenshot 2024-12-26 at 2 25 23 PM" src="https://github.com/user-attachments/assets/1988a5da-493c-4b7e-a991-2f25bfbe5625" /> |

eselkin commented:
That's odd that the label is also off on the RI. Let me check something. There was a specification of padding width that I was following by numbers not by visual reference. I'll double check. The space between labels is interesting. The middle one (on the va-select used to be the odd one out) but now it's the last one. Let me check.

eselkin commented on src/applications/facility-locator/sass/facility-locator.scss:
```diff
@@ -235,54 +285,35 @@ $facility-locator-shadow: rgba(0, 0, 0, 0.5);
-          width: 418px;
-          margin-right: 18px;
-        }
+        // width: 100%;
```
For now, maybe. The question is how the va-select will operate if they authorize the ability to extend the width to the dimensions we want.

eselkin commented on src/applications/facility-locator/components/SearchControls.jsx:
```diff
@@ -237,18 +236,21 @@ const SearchControls = props => {
+        <div className="facility-type-dropdown-block">
+          <VaSelect
+            ref={facilityTypeDropdownRef}
+            uswds
```
Yeah, good catch. Vestiges of that migration

eselkin commented on src/applications/facility-locator/components/EmergencyCareAlert.jsx:
```diff
@@ -0,0 +1,30 @@
+    return null;
+  }
+  return (
+    <VaAlert
```
I just moved it from what it was in the main app to the component without changing the type. I don't know if there was a listener attached ever. I can change to the regular va-alert though.

eselkin replied:
Changed

eselkin commented on src/applications/facility-locator/components/SearchControls.jsx:
```diff
@@ -237,18 +236,21 @@ const SearchControls = props => {
+            required
+            id="facility-type-dropdown"
+            className={showError ? 'vads-u-padding-left--1p5' : null}
+            label="Facility Type"
```
Changed

eselkin committed:
using va-alert instead of VaAlert and removed unneeded style

eselkin committed:
cleanup some styles

eselkin commented:
@randimays So the gray background padding on the search controls was affected by the `columns` class padding change that affected the facility detail pages as well (the ones that recently got moved to two different rows because of columns).

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
working spacing

eselkin committed:
Merge branch 'VACMS-19548-orientation-of-map-controls-fl' of github.com:department-of-veterans-affairs/vets-website into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
since changed padding make dimensions relatively bigger

eselkin commented:
I think I got the spacing sort of matching on all sizes for the label/button to field/select
It's a little complicated.
The padding on the gray box has been fixed.

eselkin commented:
Nevermind. The inner padding is necessary for the error states. So we have to shrink the outer padding.

randimays commented:
All looks good to me now (except for those DST issues as we discussed) 👍🏻

eselkin committed:
vaselect error uniformity with other errors

eselkin committed:
merge fix post separation of applications

eselkin commented:
@randimays I fixed the error issue on the `va-select` to be inline with the other errors and similar to the Figma. I also updated to the new main with your separation of applications PR merged. So now when it builds it should be good for yet another once over.

randimays commented:
@eselkin reviewed the inputs in their error states; looks good! 👍🏻 Still waiting on addressing those DST issues, right?

eselkin commented:
Yeah. waiting on addressing things to DST. If they don't want to provide the things we want we may be moving the button back to a regular button component so it spans correctly. I believe we'll leave the odd width VaSelect though as that's been odd since it changed a while ago.

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
VACMS - 19549 19800 conditional render services select, and loading indicator until CCP services loaded (#33784)

* get specialties
* benefits services removal
* fix widths on va-button
* Vacms 19801 - error message if ppms services get fails (fixed branch) (#33925)
* add error message
* fix focus
* add test for focus of error after PPMS services failure
* Add comment about error state on facility type change

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl

eselkin committed:
Merge branch 'main' into VACMS-19548-orientation-of-map-controls-fl
